### PR TITLE
Support MODEL DDL statements

### LINKED ISSUE #96: Incompatibility: CREATE/DROP/ALTER MODEL statements(a.k.a. Vertex AI integration) are not supported
This incompatibility is detected by #90.

## Spec

https://cloud.google.com/spanner/docs/reference/standard-sql/data-definition-language#model_statements

```
{ CREATE MODEL | CREATE OR REPLACE MODEL | CREATE MODEL IF NOT EXISTS } model_name
[INPUT ( column_list ) OUTPUT ( column_list )]
REMOTE
[OPTIONS ( model_options )]

where column_list is:
   { column_name data_type [OPTIONS ( model_column_options )] [, ... ] }

and model_column_options is:
    {
      required = { true | false }
    }

and model_options is:
    {
      endpoint = '{endpoint_address}',
      endpoints = [ '{endpoint_address}' [, ...] ],
      default_batch_size = int64_value
    }
```

https://cloud.google.com/spanner/docs/reference/standard-sql/data-definition-language#alter-model

```
ALTER MODEL [ IF EXISTS ] model_name
SET OPTIONS ( model_options )

where model_options is:
    {
      endpoint = '{endpoint_address}',
      endpoints = [ '{endpoint_address}' [, ...] ],
      default_batch_size = int64_value
    }
```

https://cloud.google.com/spanner/docs/reference/standard-sql/data-definition-language#drop-model

```
DROP MODEL [ IF EXISTS ] model_name
```

## Reproducing input
```
CREATE MODEL m1
INPUT( feature INT64 )
OUTPUT( label STRING(MAX) )
REMOTE OPTIONS (
  endpoint = '//aiplatform.googleapis.com/projects/tp/locations/tl/endpoints/schemaless'
);
```

## Current error

```
syntax error:testdata/models.test:4:8: expected pseudo keyword: DATABASE, TABLE, INDEX, UNIQUE, NULL_FILTERED, ROLE, CHANGE but: MODEL

  4:  CREATE MODEL m1
             ^~~~~
```

## Related update of this syntax

https://cloud.google.com/spanner/docs/release-notes#October_11_2022

### PR DESCRIPTION
**Pull request by apstndb**
This PR implements `CREATE MODEL`, `ALTER MODEL`, `DROP MODEL` statements.

These statements don't support named schemas, and I think it won't support because `ML.PREDICT` uses same global namespace in whole database.

https://cloud.google.com/spanner/docs/schema-and-data-model#named-schemas
https://cloud.google.com/spanner/docs/information-schema#models

Note: I didn't introduce named constant for `OrReplace` because `p.parse*()` for DDL statement is only called in `p.parseDDL()`, so I think it is not hard to read without it.

fixes #96

### CODE REVIEW

File: ast/ast.go [hunk 1]
```diff
@@ -2878,6 +2884,76 @@ type Analyze struct {
...
+//	{{.InputOutput | sqlOpt}}
+//	REMOTE
+//	{{.Options | sqlOpt}}
+type CreateModel struct {
+	// pos = Create
+	// end = Options.end || Remote + 6
+
+	Create      token.Pos // position of "CREATE" keyword
+	Remote      token.Pos // position of "REMOTE" keyword
+	OrReplace   bool
```
Comment (makenowjust) [inline ast/ast.go:2925]: Insert a newline here. (Sorry, I didn't use "suggestion" because I guess it breaks `go fmt` format.).

File: ast/ast.go [hunk 2]
```diff
@@ -2878,6 +2884,76 @@ type Analyze struct {
...
+
+// AlterModel is ALTER MODEL statement node.
+//
+//	ALTER MODEL {{if .IfExists}}IF EXISTS{{end}} {{.Name | sql}} SET {{.Options | sql}}
+type AlterModel struct {
+	// pos = Alter
+	// end = Options.end
+
+	Alter    token.Pos
+	IfExists bool
```
Comment (makenowjust) [inline ast/ast.go:2940]: Insert a newline here.

File: ast/ast.go [hunk 3]
```diff
@@ -2878,6 +2884,76 @@ type Analyze struct {
...
+
+// DropModel is DROP MODEL statement node.
+//
+//	DROP MODEL {{if .IfExists}}IF EXISTS{{end}} {{.Name | sql}}
+type DropModel struct {
+	// pos = Drop
+	// end = Name.end
+
+	Drop     token.Pos
+	IfExists bool
```
Comment (makenowjust) [inline ast/ast.go:2953]: Insert a newline here.

**Comment by apstndb** [comment]
Thank you, all newlines are added.

**Review by makenowjust — APPROVED** [review]

**Pull request merged**